Aplib.Core/Belief/Beliefset.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Linq;
+
+namespace Aplib.Core.Belief
+{
+    /// <summary>
+    /// The <see cref="Beliefset"/> class can be inherited to define a set of beliefs for an agent.
+    /// All <i>public fields</i> of type <see cref="IBelief"/> that are defined in the inheriting class
+    /// are automatically updated when calling <see cref="UpdateBeliefs"/>.
+    /// </summary>
+    public abstract class Beliefset
+    {
+        private readonly IBelief[] _beliefs;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Beliefset"/> class.
+        /// All <i>public fields</i> of type <see cref="IBelief"/> that are defined in the inheriting class
+        /// are automatically updated when calling <see cref="UpdateBeliefs"/>.
+        /// </summary>
+        protected Beliefset()
+        {
+            _beliefs =
+                GetType().GetFields()
+                .Where(field => typeof(IBelief).IsAssignableFrom(field.FieldType))
+                .Select(field => (IBelief)field.GetValue(this))
+                .ToArray();
+        }
+
+        /// <summary>
+        /// Updates all objects of type <see cref="IBelief"/> that are defined as <i>public fields</i> in <see cref="Beliefset"/>.
+        /// </summary>
+        public void UpdateBeliefs()
```

Comment:
Same as before. If we don't want the developer to update the beliefs, we may want to make this method internal.

Reply:
Unlike `UpdateBelief`, this method _could_ be made internal, but the user could need it when implementing their own version of `Agent` inheriting from `IAgent`. I think that we need to reconsider these accessibility levels sometime later.

Reply:
Great, then it's settled.